runtime/vm/ContinuationHelpers.cpp

```diff
@@ -91,43 +91,42 @@ createContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 }
 
 j9object_t
-synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuationObject, J9VMContinuation *continuation)
+synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuationObject, ContinuationState volatile *continuationStatePtr)
 {
-	uintptr_t oldContinuationState = 0;
-	uintptr_t returnContinuationState = 0;
+	ContinuationState oldContinuationState = 0;
+	ContinuationState returnContinuationState = 0;
 	do {
-		oldContinuationState = continuation->state;
-		uintptr_t newContinuationState = oldContinuationState;
+		oldContinuationState = *continuationStatePtr;
+		ContinuationState newContinuationState = oldContinuationState;
 		VM_VMHelpers::settingCarrierAndPendingState(&newContinuationState, currentThread);
-		returnContinuationState = VM_AtomicSupport::lockCompareExchange(&continuation->state, oldContinuationState, newContinuationState);
+		returnContinuationState = VM_AtomicSupport::lockCompareExchange(continuationStatePtr, oldContinuationState, newContinuationState);
 	} while (returnContinuationState != oldContinuationState);
 	Assert_VM_false(VM_VMHelpers::isPendingToBeMounted(returnContinuationState));
 	Assert_VM_Null(VM_VMHelpers::getCarrierThread(returnContinuationState));
 
 	do {
-		if (VM_VMHelpers::isConcurrentlyScanned(returnContinuationState)) {
-			/* currentThread was low tagged (GC was already in progress), but by 'or'-ing our ID, we let GC know there is a pending mount */
-
+		while (VM_VMHelpers::isConcurrentlyScanned(returnContinuationState)) {
 			PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, continuationObject);
 			internalReleaseVMAccess(currentThread);
 
 			omrthread_monitor_enter(currentThread->publicFlagsMutex);
-			while (VM_VMHelpers::isConcurrentlyScanned(continuation->state)) {
-				/* GC is still concurrently scanning the continuation(currentThread was still low tagged), wait for GC thread to notify us when it's done. */
-				omrthread_monitor_wait(currentThread->publicFlagsMutex);
-			}
+			/* GC is still concurrently scanning the continuation(currentThread was still low tagged), wait for GC thread to notify us when it's done. */
```

Comment:
remove part of the comment in brackets (talks about low level details how state is organized)